src/config/ConfigManager.cpp

```diff
@@ -1185,10 +1186,28 @@ void CConfigManager::handleWorkspaceRules(const std::string& command, const std:
 }
 
 void CConfigManager::handleSubmap(const std::string& command, const std::string& submap) {
-    if (submap == "reset")
-        m_szCurrentSubmap = "";
-    else
-        m_szCurrentSubmap = submap;
+    if (submap == "reset") {
+        // currentCategory.starts_with("submap") is here to support the old format, otherwise we should only need the first part of the condition
+        // currentCategory will be empty when we reach this code from the old submap syntax
+        if (!m_soCurrentSubmap.getHasAtLeastOneResetBinding() && currentCategory.starts_with("submap"))
+            parseError = "The reset variable is required in a submap";
```

Comment:
soooo... with the old syntax we are allowing unrecoverable submaps?

If we are adding the check, why not print it regardless of the method?

I understand with the old syntax you might wanna print a different message, but no need to discriminate